### PR TITLE
Update readme with complete instructions

### PR DESCRIPTION
**Pull request by li-kai**

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -2,3 +2,4 @@ AWS_ACCESS_KEY_ID=
 AWS_SECRET_ACCESS_KEY=
 
 GITHUB_WEBHOOK_SECRET=
+GITHUB_APP_ID=
```
Comment (taneliang) [inline .env.example:5]: Good idea

File: README.md [hunk 2]
```diff
@@ -29,28 +29,36 @@ Cotton is deployed on AWS using the Serverless framework. It is comprised of 3 l
...
-### Deployment
-
-Run `yarn deploy`. This command uses serverless to deploy Cotton to AWS.
-
-### Dev
+1. Run `cp .env.example .env` at the repo root.
+1. Set up an AWS account if you haven't.
+1. Generate `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`, following https://serverless.com/framework/docs/providers/aws/guide/credentials/.
+1. Set `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` in `.env`.
+1. Register a new GitHub App, following https://developer.github.com/apps/building-github-apps/creating-a-github-app/. Use dummy urls as required. Permission settings are as followed:
```
Comment (taneliang) [inline README.md:41]: Hmm now that I think about it, I think you can deploy to AWS (the last 2 steps) before registering the GitHub App

File: README.md [hunk 3]
```diff
@@ -29,28 +29,36 @@ Cotton is deployed on AWS using the Serverless framework. It is comprised of 3 l
...
+    - Repository contents: R&W
+    - Issues: R&W
+    - Repo metadata: R
+    - PRs: R&W
+    - Subscribe to events: Push, Issue comment, PR review, PR review comment
+1. Generate (securely, e.g. using a password manager) and set a webhook secret for the new app, and set `GITHUB_WEBHOOK_SECRET` in `.env`.
+1. Generate the private key following https://developer.github.com/apps/building-github-apps/authentication-options-for-github-apps/#generating-a-private-key.
+1. Download the key into the repo root and rename it "gh_priv_key.pem".
+1. Use the github id on the app settings page to set `GITHUB_APP_ID` in `.env`.
```
Comment (taneliang) [inline README.md]: /nit GitHub ID

File: src/config.ts [hunk 4]
```diff
@@ -1,2 +1,3 @@
-export const gitHubAppId = 10823;
+export const gitHubAppId = process.env.GITHUB_APP_ID || 10823;
```
Comment (taneliang) [inline src/config.ts]: I think we can just remove Cotton's app ID

File: src/github/auth.ts [hunk 5]
```diff
@@ -2,14 +2,15 @@ import { readFileSync } from 'fs';
...
-import { gitHubAppId } from '../config';
+import { gitHubAppId, githubWebhookSecret } from '../config';
 
 function jwtDate(date: Date) {
   return Math.floor(date.getTime() / 1000);
 }
 
 // Generate JWT for GitHub App.
 // Requires gh_priv_key.pem to be in repo root (i.e. ../..).
+// Requires github id to be in .env
```
Comment (taneliang) [inline src/github/auth.ts]: Would `GITHUB_APP_ID` be clearer? GitHub has too many IDs

**Review by taneliang — COMMENTED** [review]
Seems good. I can resolve these comments if you're busy

File: README.md (hunk 2 above)
Comment (li-kai) [reply README.md:41]: Not sure about this, could you clarify? I think it's good to set everything up before deployment, makes it clear what depends on what.

Comment (taneliang) [reply README.md:41]: Okay sure that works too. I was just thinking that if you deploy to AWS first, you'll have a URL that you can use in this step

**Pull request merged**